### DISCUSSION TITLE
Tests failing

### DISCUSSION THREAD

**Discussion by anubhav06** [osm-search/Nominatim#2640] [Q&A]
Hi, I had been working on issue #2615 

However when I ran the tests using the following, then one of the PHP test is failing:
```
cd build
make test
``` 
![0](https://user-images.githubusercontent.com/43821031/159151256-481743f8-b400-4e95-960c-7c24b9815830.png)

I have just modified the `nominatim/cli.py` file. Are the tests failing because there is no server running ?

Any help is appreciated. Thanks!

**Comment by lonvia** [comment]
@anubhav06 Please try to answer the following question: if you were the one to had to answer your own question, do you think the information provided is sufficient? If you think it is, then please outline the steps you would undertake to help. If not, please provide all the information, you think is missing.

**Reply by anubhav06** [reply]
Hey. I'm really sorry for asking silly questions, but this is my first time working on a big project like OSM.
I'm not sure what else information should I provide, can you please share that ? 

I had completed issue #2615 and ran the [tests](https://nominatim.org/release-docs/develop/develop/Development-Environment/#executing-tests), but it showed that 1 out of the 3 test was failing (as shared in screenshot).
I also tried running the [PHP Unit Tests](https://nominatim.org/release-docs/develop/develop/Testing/#php-unit-tests-testphp) and [Python Unit Tests](https://nominatim.org/release-docs/develop/develop/Testing/#python-unit-tests-testpython), but all these test were passing.

Once again sorry for not providing all the information and asking silly questions.


I manually ran Ctest with --verbose and got the following output:

```
anubhav@DESKTOP-HBDT64C:/srv/nominatim/build$ sudo ctest test --verbose
UpdateCTestConfiguration  from :/srv/nominatim/build/DartConfiguration.tcl
Parse Config file:/srv/nominatim/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/srv/nominatim/build/DartConfiguration.tcl
Parse Config file:/srv/nominatim/build/DartConfiguration.tcl
Test project /srv/nominatim/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
    Start 1: php

1: Test command: /usr/bin/phpunit "./"
1: Test timeout computed to be: 1500
1: PHPUnit 8.5.2 by Sebastian Bergmann and contributors.
1:
1: ...................E................................
1: ............  64 / 64 (100%)
1:
1: Time: 49 ms, Memory: 6.00 MB
1:
1: There was 1 error:
1:
1: 1) Nominatim\DBTest::testAgainstDatabase
1: Nominatim\DatabaseError: Failed to establish database connection:SQLSTATE[08006] [7] FATAL:  role "root" does not exist
1:
1: /srv/nominatim/Nominatim/lib-php/DB.php:44
1: /srv/nominatim/Nominatim/test/php/Nominatim/DBTest.php:150
1:
1: ERRORS!
1: Tests: 64, Assertions: 249, Errors: 1.
1/3 Test #1: php ..............................***Failed    0.07 sec
test 2
    Start 2: phpcs

2: Test command: /usr/bin/phpcs "--report-width=120" "--colors" "lib-php"
2: Test timeout computed to be: 1500
2/3 Test #2: phpcs ............................   Passed    0.55 sec
test 3
    Start 3: pylint

3: Test command: /usr/bin/pylint "nominatim"
3: Test timeout computed to be: 1500
3:
3: --------------------------------------------------------------------
3: Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
3:
3/3 Test #3: pylint ...........................   Passed   10.09 sec

67% tests passed, 1 tests failed out of 3

Total Test time (real) =  10.71 sec

The following tests FAILED:
          1 - php (Failed)
Errors while running CTest
```


The following are the available roles in postgres:
```
anubhav@DESKTOP-HBDT64C:/srv/nominatim/build$ sudo -u postgres psql
psql (12.9 (Ubuntu 12.9-0ubuntu0.20.04.1))
Type "help" for help.

postgres=# \du
                                   List of roles
 Role name |                         Attributes                         | Member of
-----------+------------------------------------------------------------+-----------
 anubhav   | Superuser, Create role, Create DB                          | {}
 postgres  | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
 www-data  |                                                            | {}

postgres=#
```

Does this mean I have to add a new role called `root` ?
